### PR TITLE
Stats in JSON format

### PR DESCRIPTION
**Pull request by jenrik**
Add support for writing stats in JSON format.

The feature is enabled by setting the `get5_stats_path_format` cvar to a value ending in `.json`

_The work is paid for in part by [ehub.gg](https://ehub.gg/)_

### CODE REVIEW

File: scripting/get5/stats.sp [hunk 1]
```diff
@@ -517,8 +517,84 @@ static int GetClutchingClient(int csTeam) {
 public void DumpToFile() {
   char path[PLATFORM_MAX_PATH + 1];
   if (FormatCvarString(g_StatsPathFormatCvar, path, sizeof(path))) {
-    g_StatsKv.ExportToFile(path);
+    DumpToFilePath(path);
+  }
+}
+
+public bool DumpToFilePath(const char[] path) {
```
Comment (splewis) [inline scripting/get5/stats.sp:524]: A suggestion:

* Add a generic `IsJSONPath` function  to util.sp
* That implementation can be simplified doing something like: `StrCompare(path[path_length - 5], ".json") == 0` (after validating path_length > 5 still)
* Replace [this](https://github.com/splewis/get5/blob/fae6a4f76eb042cb555cc8b33ed67ebb0d87cb45/scripting/get5/matchconfig.sp#L149) with a call to that function
* Make this function: `return IsJSONPath(path) : DumpToJSONFile(path) :  g_StatsKv.ExportToFile(path);`

File: scripting/get5/stats.sp [hunk 2]
```diff
@@ -517,8 +517,84 @@ static int GetClutchingClient(int csTeam) {
...
+  JSON_Object stats = EncodeKeyValue(g_StatsKv);
+  g_StatsKv.Rewind();
+
+  File stats_file = OpenFile(path, "w");
+  if (stats_file == null) {
+    LogError("Failed to open stats file");
+    return false;
   }
+
+  char jsonBuffer[8192]; // 8 KiB
```
Comment (splewis) [inline scripting/get5/stats.sp:556]: I'm not sure how large the json data is in practice TBH, I'm wondering if 8KiB is enough; do you have a sense of the size for a normal 10-person 3-map match?

Comment (jenrik) [reply scripting/get5/stats.sp:556]: This was a guess based on intuition. I tried 32 KiB but that crashed with a out of heap or stack space error (I don't recall which)

File: scripting/get5/stats.sp (hunk 1 above)
Comment (jenrik) [reply scripting/get5/stats.sp:524]: Implemented in a3e9d2b6323fa5df4fae241eb0753aadf3d4f820

**Comment by ErcouldnT** [comment]
Hi, sir

I have just started [this project](https://github.com/ErcouldnT/Get5-Match-Stats-API) about JSON match stats.

If you are able to send match stats JSON body as a POST request to a server set with get5_match_stats_api_url that would be awesome!

Erkut

**Pull request merged**